From 517ef9e3b1e8b9f023757e8271e6f0ce9ad078c5 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Mon, 09 Nov 2020 13:30:12 +0200
Subject: [PATCH] [SW-12755] gaudi: security: unprotect secured bits

Unprotect the IF RTR CTRL secured bits explicitly.
This is in preparation for dynamic security and specifically for the IF
RTR CTRL secured bits protection which we will toggle during the dynamic
protection.

Change-Id: I86a5a91faa736b0054071812b8bc48286926d25e
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index 4007501..c3b7fc1 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -591,9 +591,11 @@
 	regmap_writel(mask, PRIV_PROT_REG(prot_base));
 
 	/* offsets: 0x600 - 0x680 allowed access */
+	prot_base += SNGL_REG_PROT_SIZE;
+	regmap_writel(~0, PRIV_PROT_REG(prot_base));
 
 	/* offsets: 0x680 - 0x700 as following: */
-	prot_base += SNGL_REG_PROT_SIZE * 2;
+	prot_base += SNGL_REG_PROT_SIZE;
 	mask = (PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_PRIV_BASE_LOW_AW_0) &
 		PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_PRIV_BASE_LOW_AW_1) &
 		PROT_BIT(mmSIF_RTR_CTRL_0_RANGE_PRIV_BASE_LOW_AW_2) &
